app/Http/Controllers/Manager/RefLinksController.php

```diff
@@ -79,4 +79,16 @@ public function add(Request $request): JsonResponse
 
         return self::json($refLink, Response::HTTP_CREATED);
     }
+
+    public function delete(int $refLinkId): JsonResponse
+    {
+        if (null === ($refLink = RefLink::find($refLinkId))) {
+            throw new NotFoundHttpException(sprintf('No ref link found for id: %d', $refLinkId));
+        }
+        if ($refLink->used) {
+            throw new UnprocessableEntityHttpException('Ref');
```

Comment:
The error message could explain a bit more

Reply:
It is an error. It should be: `Referral link was used and cannot be deleted`. Fixed